lib/util/mime_utils.js

```diff
@@ -96,6 +96,47 @@ shaka.util.MimeUtils = class {
     return codecs.split(',');
   }
 
+  /**
+   * Get the normalized codec from a codec string,
+   * grouping them as long as they are compatible.
+   *
+   * @param {string} codecString
+   * @return {string}
+   */
+  static getNormalizedCodec(codecString) {
+    const parts =
+      shaka.util.MimeUtils.getCodecParts_(codecString.toLowerCase());
+    const base = parts[0];
+    const profile = parts[1];
+    switch (true) {
+      case base === 'mp4a' && profile === '66':
+      case base === 'mp4a' && profile === '67':
+      case base === 'mp4a' && profile === '68':
+        return 'mpeg2_aac';
+      case base === 'mp4a' && profile === '69':
+      case base === 'mp4a' && profile === '6b':
+        return 'mp3';
+      case base === 'mp4a' && profile === '40.2':
+      case base === 'mp4a' && profile === '40.02':
+      case base === 'mp4a' && profile === '40.5':
+      case base === 'mp4a' && profile === '40.05':
+      case base === 'mp4a' && profile === '40.29':
+        return 'mpeg4_aac';
+      case base === 'mp4a' && profile === '40.42':
+        return 'mpeg4_xhe_aac'; // Extended HE-AAC
```

Comment:
Is it true the other way though, what if the decoder supports AAC but not XHE-AAC?

Reply:
It's fine, because we filter streams individually before this point.  If there's no XHE-AAC decoder, we'll never try to connect XHE-AAC streams with anything else, because they'll be removed as unplayable.

The only way normalizing both as AAC would be a problem would be if there were two separate AAC decoders.  But that wouldn't make sense.